src/test/java/com/google/cloud/flink/bigquery/integration/IntegrationTestUtils.java

```diff
@@ -141,18 +174,30 @@ public static void createTable(String dataset, String table, String function)
 
     String base64encodedString = Base64.getEncoder().encodeToString("byte-test".getBytes("utf-8"));
 
-    if (function.equals("read")) {
-      Map<String, Object> rowContent = new HashMap<>();
-      rowContent.put("numeric_datatype", 123.345);
-      rowContent.put("string_datatype", "flink");
-      rowContent.put("bytes_datatype", base64encodedString);
-      rowContent.put("integer_datatype", 12345);
-      rowContent.put("float_datatype", 50.05f);
-      rowContent.put("boolean_datatype", true);
-      rowContent.put("timestamp_datatype", "2022-03-17 17:11:53 UTC");
-      rowContent.put("date_datatype", "2022-01-01");
-      rowContent.put("datetime_datatype", "2022-03-17T13:20:23.439071");
-      rowContent.put("geography_datatype", "POINT(51.500989020415 -0.124710813123368)");
+    Map<String, Object> subRowContent = new HashMap<>();
+    subRowContent.put("record1", "stringTest1");
+    subRowContent.put("record2", 1);
+    subRowContent.put("record3", true);
+
+    Map<String, Object> rowContent = new HashMap<>();
+    rowContent.put("numeric_datatype", 123.345);
+    rowContent.put("string_datatype", "flink");
+    rowContent.put("bytes_datatype", base64encodedString);
+    rowContent.put("integer_datatype", 12345);
+    rowContent.put("float_datatype", 50.05f);
+    rowContent.put("boolean_datatype", true);
+    rowContent.put("timestamp_datatype", "2022-03-17 17:11:53 UTC");
+    rowContent.put("date_datatype", "2022-01-01");
+    rowContent.put("datetime_datatype", "2022-03-17T13:20:23.439071");
+    rowContent.put("time_datatype", "13:20:23.439071");
+    rowContent.put("geography_datatype", "POINT(51.500989020415 -0.124710813123368)");
+    rowContent.put("record_datatype", subRowContent);
+    rowContent.put("array_datatype", new String[] {"string1", "string2", "string3"});
+    InsertAllResponse response =
+        bq.insertAll(InsertAllRequest.newBuilder(tableId).addRow(rowContent).build());
+    Thread.sleep(20000);
```

Comment:
Why do we have the sleep() here? Will customers need to add sleep() as well?

Reply:
@davidrabinowitz ,This for testing purpose to get the count and do assertions for test cases, but not required in real time to set this value. In Spark connector also sleep() is used in the test cases